src/simsopt/util/permanent_magnet_helper_functions.py

```diff
@@ -0,0 +1,716 @@
+
+    Args:
+        filename: String denoting the name of the coils file. 
+
+    Returns:
+        coils: List of CurveXYZFourier class objects.
+        base_currents: List of Current class objects.
+        ncoils: Integer representing the number of coils.
+    """
+    from simsopt.geo import CurveXYZFourier
+    from simsopt.field import Current
+
+    ncoils = np.loadtxt(filename, skiprows=1, max_rows=1, dtype=int)
+    order = int(np.loadtxt(filename, skiprows=8, max_rows=1, dtype=int))
+    coilcurrents = np.zeros(ncoils)
+    xc = np.zeros((ncoils, order + 1))
+    xs = np.zeros((ncoils, order + 1))
+    yc = np.zeros((ncoils, order + 1))
+    ys = np.zeros((ncoils, order + 1))
+    zc = np.zeros((ncoils, order + 1))
+    zs = np.zeros((ncoils, order + 1))
+    # load in coil currents and fourier representations of (x, y, z)
+    for i in range(ncoils):
+        coilcurrents[i] = np.loadtxt(filename, skiprows=6 + 14 * i, max_rows=1, usecols=1)
+        xc[i, :] = np.loadtxt(filename, skiprows=10 + 14 * i, max_rows=1, usecols=range(order + 1))
+        xs[i, :] = np.loadtxt(filename, skiprows=11 + 14 * i, max_rows=1, usecols=range(order + 1))
+        yc[i, :] = np.loadtxt(filename, skiprows=12 + 14 * i, max_rows=1, usecols=range(order + 1))
+        ys[i, :] = np.loadtxt(filename, skiprows=13 + 14 * i, max_rows=1, usecols=range(order + 1))
+        zc[i, :] = np.loadtxt(filename, skiprows=14 + 14 * i, max_rows=1, usecols=range(order + 1))
+        zs[i, :] = np.loadtxt(filename, skiprows=15 + 14 * i, max_rows=1, usecols=range(order + 1))
+
+    # CurveXYZFourier wants data in order sin_x, cos_x, sin_y, cos_y, ...
+    coil_data = np.zeros((order + 1, ncoils * 6))
+    for i in range(ncoils):
+        coil_data[:, i * 6 + 0] = xs[i, :]
+        coil_data[:, i * 6 + 1] = xc[i, :]
+        coil_data[:, i * 6 + 2] = ys[i, :]
+        coil_data[:, i * 6 + 3] = yc[i, :]
+        coil_data[:, i * 6 + 4] = zs[i, :]
+        coil_data[:, i * 6 + 5] = zc[i, :]
+
+    # Set the degrees of freedom in the coil objects
+    base_currents = [Current(coilcurrents[i]) for i in range(ncoils)]
+    ppp = 20
+    coils = [CurveXYZFourier(order*ppp, order) for i in range(ncoils)]
+    for ic in range(ncoils):
+        dofs = coils[ic].dofs_matrix
+        dofs[0][0] = coil_data[0, 6*ic + 1]
+        dofs[1][0] = coil_data[0, 6*ic + 3]
+        dofs[2][0] = coil_data[0, 6*ic + 5]
+        for io in range(0, min(order, coil_data.shape[0]-1)):
+            dofs[0][2*io+1] = coil_data[io+1, 6*ic + 0]
+            dofs[0][2*io+2] = coil_data[io+1, 6*ic + 1]
+            dofs[1][2*io+1] = coil_data[io+1, 6*ic + 2]
+            dofs[1][2*io+2] = coil_data[io+1, 6*ic + 3]
+            dofs[2][2*io+1] = coil_data[io+1, 6*ic + 4]
+            dofs[2][2*io+2] = coil_data[io+1, 6*ic + 5]
+        coils[ic].local_x = np.concatenate(dofs)
+    return coils, base_currents, ncoils
+
+
+def coil_optimization(s, bs, base_curves, curves, OUT_DIR):
```

Comment:
Please use lowercase for OUT_DIR in all places. Also, think about supplying a default argument such as None to use the current directory.